src/marks/difference.js

```diff
@@ -0,0 +1,109 @@
+import {area as shapeArea, line as shapeLine} from "d3";
+import {create} from "../context.js";
+import {maybeCurve} from "../curve.js";
+import {Mark} from "../mark.js";
+import {identity, indexOf, isColor} from "../options.js";
+import {applyIndirectStyles, applyTransform, getClipId} from "../style.js";
+import {maybeDenseIntervalX} from "../transforms/bin.js";
+
+const defaults = {
+  ariaLabel: "difference",
+  fill: "none",
+  stroke: "currentColor",
+  strokeWidth: 1.5,
+  strokeLinecap: "round",
+  strokeLinejoin: "round",
+  strokeMiterlimit: 1
+};
+
+function maybeColor(value) {
+  if (value == null) return "none";
+  if (!isColor(value)) throw new Error(`invalid color: ${value}`);
+  return value;
+}
+
+class Difference extends Mark {
+  constructor(data, options = {}) {
+    const {x1, y1, x2, y2, curve, tension, positiveColor = "green", negativeColor = "blue"} = options;
+    super(
+      data,
+      {
+        x1: {value: x1, scale: "x"},
+        y1: {value: y1, scale: "y"},
+        x2: {value: x2, scale: "x", optional: true},
+        y2: {value: y2, scale: "y", optional: true}
+      },
+      options,
+      defaults
+    );
+    this.curve = maybeCurve(curve, tension);
+    this.positiveColor = maybeColor(positiveColor);
+    this.negativeColor = maybeColor(negativeColor);
+  }
+  filter(index) {
+    return index;
+  }
```

Comment:
I don't understand why we're not using the default filter; if any point's y1 or y2 is invalid, the whole area disappears.

Reply:
I copied this from area/line. I think the idea is that it uses the _line_.defined and _area_.defined check instead to interrupt the path, rather pretending the point doesn’t exist and connecting across the gap. I’ll add some tests to make sure this works.

Reply:
The filtering is handled by groupIndex, which I’ve now restored.